website/static/js/nodesPrivacy.js

```diff
@@ -0,0 +1,280 @@
+/**
+ * Controller for changing privacy settings for a node and its children.
+ */
+'use strict';
+
+var $ = require('jquery');
+var ko = require('knockout');
+var bootbox = require('bootbox');
+var Raven = require('raven-js');
+
+var oop = require('./oop');
+var $osf = require('./osfHelpers');
+var Paginator = require('./paginator');
+var osfHelpers = require('js/osfHelpers');
+var m = require('mithril');
+var Treebeard = require('treebeard');
+var NodesPrivacyTreebeard = require('js/nodesPrivacySettingsTreebeard');
+
+var MESSAGES = {
+    makeProjectPublicWarning:
+    'Please review your projects, components, and add-ons for sensitive or restricted information before making them public.' +
+    '<br><br>Once they are made public, you should assume they will always be public. You can ' +
+    'return them to private later, but search engines (including Google’s cache) or others may access files before you do.',
+
+    selectNodes: 'Adjust your privacy settings by checking the boxes below. ' +
+    '<br><br><b>Checked</b> projects and components will be <b>public</b>.  <br><b>Unchecked</b> components will be <b>private</b>.',
+    confirmWarning: {
+        nodesPublic: 'The following projects and components will be made <b>public</b>.',
+        nodesPrivate: 'The following projects and components will be made <b>private</b>.',
+        nodesNotChangedWarning: 'No nodes were changed.'
+    }
+};
+
+/**
+ * take treebeard tree structure of nodes and get a dictionary of parent node and all its
+ * children
+ */
+function getNodesOriginal(nodeTree, nodesOriginal) {
+    var i;
+    var nodeId = nodeTree.node.id;
+    nodesOriginal[nodeId] = {
+        public: nodeTree.node.is_public,
+        id: nodeTree.node.id,
+        title: nodeTree.node.title,
+        canWrite: nodeTree.node.can_write,
+        changed: false
+    };
+
+    if (nodeTree.children) {
+        for (i in nodeTree.children) {
+            nodesOriginal = getNodesOriginal(nodeTree.children[i], nodesOriginal);
+        }
+    }
+    return nodesOriginal;
+}
+
+/**
+ * patches all the nodes in a changed state
+ * uses API v2 bulk requests
+ */
+function patchNodesPrivacy(nodes) {
+    var nodesV2Url = window.contextVars.apiV2Prefix + 'nodes/';
+    var nodesPatch = nodes.splice(0,10).map(function (node) {
```

Comment:
Oh, I see, you modify the list in place then send nodes 10 at a time.

It would preferable to avoid modifying the input list (what if other code depends on it?), but I think can happen in a separate PR.